### ISSUE TITLE
Calling notifyError with a nil error throws NSInvalidArgumentException

### ISSUE THREAD

**Issue by combinatorial** [bugsnag/bugsnag-cocoa#461]
### Description
We have some rare cases where we see crashes due to passing a nil error to notifyError. It seems like the SDK could guard against this and not crash?

### Issue
```
NSInvalidArgumentException: *** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[1]

0  CoreFoundation             ___exceptionPreprocess
1  libobjc.A.dylib            _objc_exception_throw
2  CoreFoundation             __CFThrowFormattedException
3  CoreFoundation             -[__NSPlaceholderDictionary initWithObjects:forKeys:count:].cold.4
4  CoreFoundation             -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]
5  CoreFoundation             +[NSDictionary dictionaryWithObjects:forKeys:count:]
6  Concepts                   __37-[BugsnagNotifier notifyError:block:]_block_invoke (BugsnagNotifier.m:541:41)
7  Concepts                   -[BugsnagNotifier notify:handledState:block:] (BugsnagNotifier.m:654:9)
8  Concepts                   -[BugsnagNotifier notifyError:block:] (BugsnagNotifier.m:537:5)
9  Concepts                   +[Bugsnag notifyError:] (Bugsnag.m:129:9)
```

### Environment

Library versions:

* Bugsnag version (from your Podfile, Podfile.lock, or elsewhere):Bugsnag (5.23.0)
* CocoaPods version (if any) (`pod -v`): 1.8.4
* iOS/tvOS/macOS version(s): iOS 13.3
* debug mode or production?: production

### Example code snippet

```objc
[Bugsnag notifyError:nil];
```

**Comment by phillipsam** [comment]
Hi @combinatorial thanks for the report! We will look into a fix for this in an upcoming release.

**Comment by steve-nester-uk** [comment]
Fixed in [v6.1.0](https://github.com/bugsnag/bugsnag-cocoa/releases/tag/v6.1.0) / PLAT-3988
